nixos/modules/services/x11/window-managers/dk.nix

```diff
@@ -0,0 +1,36 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.xserver.windowManager.dk;
+in
+
+{
+  options = {
+    services.xserver.windowManager.dk = {
+      enable = mkEnableOption (lib.mdDoc "dk");
+
+      package = mkOption {
+        type        = types.package;
+        default     = pkgs.dk;
+        defaultText = literalExpression "pkgs.dk";
+        description = lib.mdDoc ''
+          dk package to use.
+        '';
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    services.xserver.windowManager.session = singleton {
+      name  = "dk";
+      start = ''
+        export _JAVA_AWT_WM_NONREPARENTING=1
```

Comment:
What is the relationship with Java??

Reply:
@AndersonTorres some Java application ignore wm hints. This variable controls this, and it is commonly specified in wayland compositors, but some X WMs like bspwm require it too